pkg/proxy/service.go

```diff
@@ -341,11 +344,11 @@ func (sm *ServiceMap) apply(changes *ServiceChangeTracker, UDPStaleClusterIP set
 	changes.lock.Lock()
 	defer changes.lock.Unlock()
 	for _, change := range changes.items {
-		sm.merge(change.current)
+		sm.merge(change.current, changes.serviceMapEntryChanged)
 		// filter out the Update event of current changes from previous changes before calling unmerge() so that can
 		// skip deleting the Update events.
 		change.previous.filter(change.current)
-		sm.unmerge(change.previous, UDPStaleClusterIP)
+		sm.unmerge(change.previous, UDPStaleClusterIP, changes.serviceMapEntryChanged)
```

Comment:
Similar question here, is it possible to rework this to call a single function with all the changes here instead? Maybe something like `changes.onServiceChange(change)`.

Reply:
This will be similar to the above change in endpoints